### PR TITLE
A couple improvements to `EventTarget` event firing.

### PR DESCRIPTION
**Pull request by frewsxcv**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14036)

### CODE REVIEW

**Comment by KiChjang** [comment]
Looks like this fixes #13884.

**Comment by KiChjang** [comment]
Oh wait, no, this doesn't look like it's following the language from the spec update. Do you mind editing this PR or making a new one to properly address #13884?

**Comment by frewsxcv** [comment]
Not too interested in that issue, so I'd prefer not to change this PR :)

**Comment by frewsxcv** [comment]
Blocked on https://github.com/servo/servo/pull/14043.

**Comment by frewsxcv** [comment]
Just rebased this off @SimonSapin's changes in #14043. I actually didn't need to do https://github.com/servo/html5ever/issues/230 since event names (from what I understand) will now go in the `servo_atoms` crate.

File: components/script/dom/event.rs [hunk 1]
```diff
@@ -330,7 +330,9 @@ impl Runnable for EventRunnable {
 
     fn handler(self: Box<EventRunnable>) {
         let target = self.target.root();
-        target.fire_event_with_params(&*self.name, self.bubbles, self.cancelable);
+        let bubbles = self.bubbles.clone();
+        let cancelable = self.cancelable.clone();
```
Comment (Ms2ger) [inline components/script/dom/event.rs]: What's this about?

Comment (nox) [inline components/script/dom/event.rs]: Oops, filed https://github.com/servo/servo/issues/14062.

**Pull request merged**